libs/cardano-ledger-core/src/Cardano/Ledger/Core/Era.hs

```diff
@@ -130,6 +132,8 @@ type family EraRule (rule :: Symbol) era = (r :: Type) | r -> rule
 -- does not provide for us unfortunately.
 type family EraRuleFailure (rule :: Symbol) era = (r :: Type) | r -> rule era
 
+type family EraRuleEvent (rule :: Symbol) era = (r :: Type) | r -> rule era
```

Comment:
Sorry if it's obvious, but could you please explain what exactly having this type family enables use to do?  Is it related to improving the test framework?

Reply:
The problem is that `STS` isn't aware of the `era` type parameter, so you can't make `Event` injective on `rule era`. Hopefully we can do some refactoring in the future to make an era-aware `STS` class.

Reply:
And what do you need the injectivity for? Is it about reusing events in other rules/eras?